### PR TITLE
EN-2205 EN-2206 (5.0.1) Moved kie form overrides management inside BpmFormWidget

### PR DESCRIPTION
**Pull request by zonia3000**

### CODE REVIEW

**Comment by entando-service** [comment]
All is Well

File: plugins/entando-plugin-jpkiebpm/src/main/db/update/scriptServ.sql [hunk 1]
```diff
@@ -17,3 +17,7 @@ WITH (
 );
 ALTER TABLE public.jpkiebpm_kieformoverride
   OWNER TO agile;
+
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/db/update/scriptServ.sql]: We need to capture this in migration/porting instructions on confluence

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/bpmwidgetinfo/BpmWidgetInfoManager.java [hunk 2]
```diff
@@ -154,29 +150,44 @@ public void setBpmWidgetInfoDAO(IBpmWidgetInfoDAO bpmWidgetInfoDAO) {
...
                 event.getEventType().equals(PageChangedEvent.EVENT_TYPE_REMOVE_WIDGET)) {
             List<BpmWidgetInfo> bpmWidgetInfoList = this.getBpmWidgetInfoDAO().searchBpmWidgetInfo(event.getPage().getCode(), event.getFramePosition());
             if (null != bpmWidgetInfoList) {
 
                 for (final BpmWidgetInfo bpmWidgetInfo : bpmWidgetInfoList) {
-
-                    bpmWidgetInfo.setInformationDraft(null);
-                    bpmWidgetInfo.setFramePosDraft(null);
-                    this.getBpmWidgetInfoDAO().updateBpmWidgetInfo(bpmWidgetInfo);
+                    if(bpmWidgetInfo.getInformationOnline() == null) {
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/bpmwidgetinfo/BpmWidgetInfoManager.java]: Are we sure there is no false version of this? Always null?

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/apsadmin/portal/specialwidget/BpmFormWidgetAction.java [hunk 3]
```diff
@@ -108,10 +123,20 @@ protected void setPropertyWidget(final List<WidgetTypeParameter> parameters, fin
 
     @Override
     protected void createValuedShowlet() throws Exception {
-        Widget widget = this.createNewWidget();
+        // If the widget is not new we need to retrieve the widgetInfoId but not
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/apsadmin/portal/specialwidget/BpmFormWidgetAction.java]: Very nice! Great comment!

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/apsadmin/portal/specialwidget/BpmFormWidgetAction.java [hunk 4]
```diff
@@ -354,36 +379,54 @@ protected void initSharedParameters(final BpmWidgetInfo widgetInfo) {
...
         } catch (ApsSystemException e) {
             logger.error("Error loading WidgetInfo", e);
         }
     }
 
     public String chooseForm() {
         try {
-            //System.out.println("nothing to do");
+            overrides = new ArrayList<>();
+            // TODO: load existing
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/apsadmin/portal/specialwidget/BpmFormWidgetAction.java]: Remove comment or resolve TODO

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/apsadmin/portal/specialwidget/BpmFormWidgetAction.java [hunk 5]
```diff
@@ -394,48 +437,47 @@ public String chooseForm() {
...
-            String containerId = params[1];
-            EntitySearchFilter processIdFilter = new EntitySearchFilter("processId", false, processId, true);
-            EntitySearchFilter containerIdFilter = new EntitySearchFilter("containerId", false, containerId, true);
-            filters = this.addFilter(filters, processIdFilter);
-            filters = this.addFilter(filters, containerIdFilter);
-        }
-        return filters;
-    }
-
+//    public Map<String, KieFormOverride> getFormOverridesMap() throws ApsSystemException {
```
Comment (joewhite101) [inline plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/apsadmin/portal/specialwidget/BpmFormWidgetAction.java]: Remove commented out code if not needed

File: plugins/entando-plugin-jpkiebpm/src/main/db/update/scriptServ.sql (hunk 1 above)
Comment (zonia3000) [reply plugins/entando-plugin-jpkiebpm/src/main/db/update/scriptServ.sql]: Done. I had also forgot another column...

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/bpmwidgetinfo/BpmWidgetInfoManager.java (hunk 2 above)
Comment (zonia3000) [reply plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/bpmwidgetinfo/BpmWidgetInfoManager.java]: What to you mean by "no false version of this"?
If this variable is null it means that the widget is present only in the draft version of the page. So if it is removed we have to remove also the related widget information and form overrides.

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/apsadmin/portal/specialwidget/BpmFormWidgetAction.java (hunk 3 above)
Comment (zonia3000) [reply plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/apsadmin/portal/specialwidget/BpmFormWidgetAction.java]: thanks!

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/apsadmin/portal/specialwidget/BpmFormWidgetAction.java (hunk 5 above)
Comment (zonia3000) [reply plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/apsadmin/portal/specialwidget/BpmFormWidgetAction.java]: Yes, I forgot this before committing.

File: plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/apsadmin/portal/specialwidget/BpmFormWidgetAction.java (hunk 4 above)
Comment (zonia3000) [reply plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/apsadmin/portal/specialwidget/BpmFormWidgetAction.java]: Ok, checked, it is not necessary. If you change the process the old overrides are removed when you click on save. There is nothing to load.

**Pull request merged**